php-sync/sfs-sync.php

```diff
@@ -305,40 +306,32 @@ public function scheduleBatches ($mode, $toSchedule) {
 
 			$dir = $this->config["BATCHDIR"]."/$mode/$node";
 			if (!is_dir ($dir)) {
-				continue;
+				mkdir ($dir, 0777, TRUE);
 			}
-			
-			$hBatchDir = opendir($dir);
-			if ($hBatchDir === FALSE) {
+			if(!is_dir($dir)){
 				syslog(LOG_CRIT, "Cannot opendir $dir, will retry in ".$this->config["FAILTIME"]." seconds");
 				$this->setFailing ($node);
-				$tasks[] = array ();
+				continue;
+			}
+
+			$allBatches = glob($dir . '/*_*.batch');
```

Comment:
yeah ok, if you assume to have that much files lying around.ok. I didn't expect to have many files lying around if the service is running. It looked easier for me.